### PR TITLE
Fix / usermenu item count & correct html structure

### PR DESCRIPTION
**Pull request by langemike**
This change announced the correct amount of list items in the user menu. I also improved wrapped the sidebar menu items in a list, so a screen reader user knows how may items there are in the menu.

Originally I wanted to go for a border-bottom styling instead of the suggested `<li role="presentation"><hr/></li>`, but it took too much time to get the styling identical.

I chosen a `h2` for the heading because, we should limit the `h1` to one. And using a `h3` is illogical (because it is missing the `h2`). I could not find an better solution within the time. I'm open for ideas, if you disagree with the approach.

Ticket: https://videodock.atlassian.net/browse/OTT-842

### CODE REVIEW

File: packages/ui-react/src/components/UserMenu/UserMenu.tsx
Comment (MelissaDTH) [inline packages/ui-react/src/components/UserMenu/UserMenu.tsx:69]: We've internally discussed that we will not merge this change/PR yet, because currently `inert` is not correctly applied yet, causing the backdrop to still be accessible/scrollable and therefore the situation arises where this `<h2>` is in conflict with the overall HTML structure. If the `inert` of the sidebar is fixed (one of my tickets), this change can be applied.

Comment (langemike) [reply packages/ui-react/src/components/UserMenu/UserMenu.tsx:69]: The h2 somehow doesn't feel "right", but it seems like the best approach here. I checked with Axe devtools and it didn't complain. The h2 is enclosed in a `<section>`. So it should be alright.

**Comment by langemike** [comment]
axe devools complains about `<li role="presentation"><hr/></li>` though... that is really unfortunate because that is the only issue it complains about. I guess I have to spend some time on the border-bottom.... 

Result:

Screenshot 2024-02-15 at 17 15 28

**Comment by langemike** [comment]
Fixed the above axe validation error by replacing `<li role="presentation"><hr/></li>` with CSS

File: packages/ui-react/src/containers/Layout/Layout.tsx
Comment (royschut) [inline packages/ui-react/src/containers/Layout/Layout.tsx]: I know this code is not introduced by you, but are these dynamic tabIndexes needed? The whole Sidebar gets `inert` when `isOpen === false`.

File: packages/ui-react/src/components/UserMenu/UserMenu.tsx
Comment (royschut) [reply packages/ui-react/src/components/UserMenu/UserMenu.tsx:69]: It's a header of the section, and level 2 seems the only appropriate, so to me it seems the best option.

File: packages/ui-react/src/containers/Layout/Layout.tsx
Comment (royschut) [inline packages/ui-react/src/containers/Layout/Layout.tsx]: Don't we want to replace this one with your solution as well?

**Review by royschut — APPROVED** [review]
Nice Mike!
Few minors:

File: packages/ui-react/src/components/UserMenu/UserMenu.tsx
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/UserMenu/UserMenu.tsx:74]: Inside the `<ProfilesMenu />`, there are also list items and a divider 😅

File: packages/ui-react/src/containers/Layout/Layout.tsx
Comment (langemike) [reply packages/ui-react/src/containers/Layout/Layout.tsx]: I believe I kept this there when implementing `inert` for the sidebar, because you could come in to a state were you sidebar was closed and you were still able to focus the hidden buttons. Potentially due a screen reader bug.

But theoretically that shouldn't occur. So I reevaluated it, tested it again on iOS/Android and it seems like it work. So I committed my change. In the worst case, we can always revert it if this issue reappears during testing.

File: packages/ui-react/src/containers/Layout/Layout.tsx
Comment (langemike) [reply packages/ui-react/src/containers/Layout/Layout.tsx]: I intentionally skipped  this `<hr>` due to time and I do not consider it a requirement (or something we could fail upon during an audit).  But since you mention it, I took the effort and fixed it ;)

File: packages/ui-react/src/components/UserMenu/UserMenu.tsx
Comment (langemike) [reply packages/ui-react/src/components/UserMenu/UserMenu.tsx:74]: I think i will need another 2 hours to optimize the `ProfilesMenu`. It also got a heading and a divider.
I want to split them in to 2 lists with each their own heading. But it also got `onFocus`/`onBlur` handlers for the fly-out menu, which I want to reevaluate after restructure, if they work correctly.

Can you tell me an appconfig ID which has profiles enabled? With your approval, I will fix it separately.

Comment (ChristiaanScheermeijer) [reply packages/ui-react/src/components/UserMenu/UserMenu.tsx:74]: The profiles feature has been enabled, but we can do it separately from this PR 👍

**Review by ChristiaanScheermeijer — APPROVED** [review]

**Comment by ChristiaanScheermeijer** [comment]
Oh, snapshots are failing I think

**Review by MelissaDTH — APPROVED** [review]

**Pull request merged**
